Intersect.Server/Entities/Events/EventPageInstance.cs

```diff
@@ -372,340 +373,339 @@ protected override bool ProcessMoveRoute(Player forPlayer, long timeMs)
+                        {
+                            if (mPathFinder.GetTarget() == null ||
+                                mPathFinder.GetTarget().TargetMapId != forPlayer.MapId ||
+                                mPathFinder.GetTarget().TargetX != forPlayer.X ||
+                                mPathFinder.GetTarget().TargetY != forPlayer.Y)
                             {
-                                if (mPathFinder.GetTarget() == null ||
-                                    mPathFinder.GetTarget().TargetMapId != forPlayer.MapId ||
-                                    mPathFinder.GetTarget().TargetX != forPlayer.X ||
-                                    mPathFinder.GetTarget().TargetY != forPlayer.Y)
-                                {
-                                    mPathFinder.SetTarget(
-                                        new PathfinderTarget(forPlayer.MapId, forPlayer.X, forPlayer.Y, forPlayer.Z)
-                                    );
-                                }
+                                mPathFinder.SetTarget(
+                                    new PathfinderTarget(forPlayer.MapId, forPlayer.X, forPlayer.Y, forPlayer.Z)
+                                );
+                            }
 
-                                //Todo check if next to or on top of player.. if so don't run pathfinder.
-                                if (mPathFinder.Update(timeMs) == PathfinderResult.Success)
+                            //Todo check if next to or on top of player.. if so don't run pathfinder.
+                            if (mPathFinder.Update(timeMs) == PathfinderResult.Success)
+                            {
+                                var pathDir = mPathFinder.GetMove();
+                                if (pathDir > Direction.None && CanMoveInDirection(pathDir, out var blockerType, out _))
                                 {
-                                    var pathDir = mPathFinder.GetMove();
-                                    if (pathDir > Direction.None)
+                                    if (blockerType == MovementBlockerType.NotBlocked)
+                                    {
+                                        Move(pathDir, forPlayer);
+                                        moved = true;
+                                    }
+                                    else
                                     {
-                                        if (CanMove(pathDir) == -1)
-                                        {
-                                            Move(pathDir, forPlayer);
-                                            moved = true;
-                                        }
-                                        else
-                                        {
-                                            mPathFinder.PathFailed(timeMs);
-                                        }
+                                        mPathFinder.PathFailed(timeMs);
```

Comment:
This line literally can't be hit now. When `CanMoveInDirection()` returns true `blockerType` is always going to be `NotBlocked`. The `if (blockerType == MovementBlockerType.NotBlocked)` isn't needed and the else should be moved one level up.

Reply:
👍🏽 

```csharp
                            var pathDir = mPathFinder.GetMove();
                            if (pathDir > Direction.None && CanMoveInDirection(pathDir, out _, out _))
                            {
                                Move(pathDir, forPlayer);
                                moved = true;
                            }
                            else
                            {
                                mPathFinder.PathFailed(timeMs);
                            }
```

Reply:
This should be `pathDir != Direction.None` rather than `>`, there is no guarantee the tombstone value won't get moved.

Reply:
You can also just do `CanMoveInDirection(pathDir)`? That overload exists so you don't have to do `out _, out _`